hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -342,16 +343,22 @@ void decodeInput(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
 	@SuppressWarnings("rawtypes")
 	protected Bytes computeInternal(final MessageFrame frame) {
 		final var updater = (AbstractLedgerWorldUpdater) frame.getWorldUpdater();
+		this.sideEffectsTracker = sideEffectsFactory.get();
+		updater.setSideEffectsTracker(sideEffectsTracker);
 		final var ledgers = updater.wrappedTrackingLedgers();
 
 		Bytes result;
 		ExpirableTxnRecord.Builder childRecord;
 		try {
-			childRecord = precompile.run(frame, ledgers);
+			final List<FcAssessedCustomFee> assessedCustomFees = precompile.run(frame, ledgers);
```

Comment:
It's a little odd to have the generic `run()` method for a precompile returning a `List<FcAssessedCustomFee>`, since most precompiles have nothing to do with custom fees.

Maybe instead we add a `Precompile.configureSuccessRecord(ExpirableTxnRecord.Builder childRecord)` method, and call it after `creator.createSuccessfulSyntheticRecord()`?